src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -142,22 +160,139 @@ class ActivityFeed extends React.Component<Props, State> {
+            params: {
+                filter_term: searchStr
+            },
+            successCallback,
+            errorCallback: this.errorCallback
+        });
+    }
+
+    /**
+     * File @mention contacts fetch success callback
+     *
+     * @private
+     * @param {API} api - Box API instance
+     * @param {string} searchStr - Search string to filter file collaborators by
+     * @param {Function} successCallback - Fetch success callback
+     * @return {void}
+     */
     getApproverSelectorContacts = (searchStr: string): void => {
-        // using v2 api, search for approver on file with searchStr
-        // use collaborators endpoint /files/ID/collaborators
-        // update contacts state 'approverSelectorContacts'
-        // call user passed in handlers.contacts.getApproverWithQuery, if it exists
-        const getApproverWithQuery = getProp(this.props, 'handlers.contacts.getApproverWithQuery', noop);
-        this.setState({ approverSelectorContacts: getApproverWithQuery(searchStr) });
+        if (this.approverSelectorTimeout) {
+            clearTimeout(this.approverSelectorTimeout);
+            this.approverSelectorTimeout = null;
+
+            if (this.approverCollabsAPI) {
+                this.approverCollabsAPI.destroy();
+            }
+        }
+
+        const api = getProp(this.props, 'api', null);
+        this.approverCollabsAPI = api.getFileCollaboratorsAPI();
+
+        this.approverSelectorTimeout = setTimeout(() => {
+            this.fetchFileCollaborators(this.approverCollabsAPI, searchStr, this.getApproverContactsSuccessCallback);
+        }, 2);
     };
 
+    /**
+     * File @mention contacts fetch success callback
+     *
+     * @private
+     * @param {string} searchStr - Search string to filter file collaborators by
+     * @return {void}
+     */
     getMentionSelectorContacts = (searchStr: string): void => {
-        // using v2 api, search for mention on file with searchStr
-        // use collaborators endpoint /files/ID/collaborators
-        // update contacts state 'mentionSelectorContacts'
-        // call user passed in handlers.contacts.getMentionWithQuery, if it exists
-        const getMentionWithQuery = getProp(this.props, 'handlers.contacts.getMentionWithQuery', noop);
-        this.setState({ mentionSelectorContacts: getMentionWithQuery(searchStr) });
+        if (this.mentionSelectorTimeout) {
+            clearTimeout(this.mentionSelectorTimeout);
+            this.mentionSelectorTimeout = null;
+
+            if (this.mentionCollabsAPI) {
+                this.mentionCollabsAPI.destroy();
+            }
+        }
+
+        const api = getProp(this.props, 'api', null);
+        this.mentionCollabsAPI = api.getFileCollaboratorsAPI();
+
+        this.mentionSelectorTimeout = setTimeout(() => {
+            this.fetchFileCollaborators(this.mentionCollabsAPI, searchStr, this.getMentionContactsSuccessCallback);
+        }, 2);
```

Comment:
Why a 2ms timeout here? Seems like a very small amount of time. Are you trying to accomplish debouncing?

Reply:
Just realized there's an actual way to debounce this... Updating the timeout to something realistic. This was something I used just for testing